### PR TITLE
Fix #88: Update rootCid when writing to webnative

### LINKED ISSUE #88: Bug: Model's fileSystemCid doesn't update when saving a text file on a public path
The value behind `model.fileSystemCid` doesn't reflect the up-to-date value of the public DNS link, as it is shown in e.g. the console.

The impact this has is that right-clicking a public file that was just edited will copy the wrong link to the clipboard. Opening this link will show the old version of the file, instead of the newer version.

Possible fix: Somehow update the fileSystemCid when a file is saved.

Personal thoughts: Maybe the port that handles `writeItemUtf8` should send some message back to elm. At the moment, it seems like `gotDirectoryList` is what is usually sent back to the Elm side everytime the filesystem changed. I should do the same. The `write` function should be managed by the common `exe` way of registering drive functions.
(Note to self: `fs.publish()` returns the new CID.).

### PR DESCRIPTION
**Pull request by matheus23**

### CODE REVIEW

File: src/Javascript/fs.js [hunk 1]
```diff
@@ -41,10 +41,13 @@ export async function add({ blobs, pathSegments }) {
   return await listDirectory({ pathSegments })
 }
 
-export async function write({ pathSegments, text }) {
+const encoder = new TextEncoder()
```
Comment (icidasset) [inline src/Javascript/fs.js:44]: Should we put this in `writeItemUtf8` just to be safe?
I don't understand why they made this a class 😅

File: src/Javascript/fs.js [hunk 2]
```diff
@@ -41,10 +41,13 @@ export async function add({ blobs, pathSegments }) {
...
+
+export async function writeItemUtf8({ pathSegments, text }) {
   const path = prefixedPath(pathSegments)
-  await fs.add(path, text)
+  await fs.add(path, encoder.encode(text))
   await fs.publish()
+  return await listDirectory({ pathSegments: pathSegments.slice(0, -1) })
```
Comment (icidasset) [inline src/Javascript/fs.js:50]: Just out of curiousity, what's the reason we need to do a list here?

**Review by icidasset — APPROVED** [review]
Few questions, looks good otherwise 👌

File: src/Javascript/fs.js (hunk 2 above)
Comment (icidasset) [reply src/Javascript/fs.js:50]: Oh this fixes the actual issue, ignore me 😂

File: src/Javascript/fs.js (hunk 1 above)
Comment (matheus23) [reply src/Javascript/fs.js:44]: I wanted to only 'allocate' this class once for performance/memory reasons, but I guess that's premature optimization.

Yep. Super annoying that this has to be a class. Not sure whether it actually stores any state. It definitely shouldn't.

**Pull request merged**